### PR TITLE
Playback 2023: entry point

### PR DESCRIPTION
**Pull request by leandroalonso**
| 📘 Part of: #1142 | 🎨 Designs: [Figma](https://www.figma.com/file/lH66LwxxgG8btQ8NrM0ldx/End-of-Year?type=design&node-id=1599-20385&mode=design) |
|:---:|:---:|

Updates the logic and visuals for the Playback 2023 modal, profile badge, and profile card.

| Modal | Badge | Card |
| ------ | ------ | ----- |
| Simulator Screenshot - iPhone 15 - 2023-10-16 at 13 20 43 | Simulator Screenshot - iPhone 15 - 2023-10-16 at 13 19 18 | Simulator Screenshot - iPhone 15 - 2023-10-16 at 13 19 26 |

## To test

1. Make sure you're logged in to an account that has a few episodes listened
2. Go to Profile > Settings > Beta Features > enable `endOfYear`
3. Restart the app
4. ✅ A modal should be shown
5. ✅  Tapping on "View My 2023" shows the stories
6. Dismiss it
7. ✅ There should be a badge on the "Profile" tab
8. Tap it
9. ✅ The badge should disappear
10. ✅ Check that there's a card on Profile prompting you to see your 2023
11. Tap it
12. ✅ Playback 2023 stories should start

If you want to reset the modal being shown and badge, go to Profile > Settings > Developer > Reset modal/profile badge

Please test on a big device (iPhone 15 Pro Max, for example), a normal device (iPhone 15), and a smaller device (iPod touch). Also on light/dark mode.

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

File: podcasts/End of Year/Views/EndOfYearModal.swift [hunk 1]
```diff
@@ -5,25 +5,24 @@ struct EndOfYearModal: View {
...
-
-                cover
+                Image("modal_cover")
+                    .resizable()
+                    .scaledToFit()
+                    .clipShape(RoundedRectangle(cornerRadius: 16))
 
                 Text(L10n.eoyDescription)
-                    .font(style: .body, maxSizeCategory: .accessibilityMedium)
+                    .font(style: .body, weight: .medium, maxSizeCategory: .accessibilityMedium)
```
Comment (emilylaguna) [inline podcasts/End of Year/Views/EndOfYearModal.swift]: The font size is a bit too big, and the linespacing a bit tight when compared to the designs. 


Screenshot 2023-10-19 at 1 54 55 AM

**Review by emilylaguna — CHANGES_REQUESTED** [review]
Nice work. Left some comments. 

Also, I noticed that if I'm logged out and I tap go to open the EOY I am brought to the old login view.

File: podcasts/End of Year/Views/EndOfYearModal.swift (hunk 1 above)
Comment (leandroalonso) [reply podcasts/End of Year/Views/EndOfYearModal.swift]: Changed in 397f81eccb89bc2590ee94ffaabfd3d514c075fc

**Comment by leandroalonso** [comment]
> Also, I noticed that if I'm logged out and I tap go to open the EOY I am brought to the old login view.

This is already a task in #1142 :)

**Review by emilylaguna — CHANGES_REQUESTED** [review]
On the iPad the sheet is clipped in both landscape and portrait.

**Comment by leandroalonso** [comment]
@emilylaguna fixed in 6f790846f26b509ed0770bd2fb07513394ead5fd

**Review by emilylaguna — APPROVED** [review]

**Pull request merged**